app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -29,10 +29,25 @@ def endpoint
   end
 
   def command_headers
-    hours = (registration_params['jwt_expiration'].presence || 4).to_i.hours.to_i
-    scope = [{ controller: :registration, actions: [:global, :host] }]
-    jwt = User.current.jwt_token!(expiration: hours, scope: scope)
+    jwt_args = {
+      scope: [{ controller: :registration, actions: [:global, :host] }],
+    }
 
-    "-H 'Authorization: Bearer #{jwt}'"
+    if registration_params['jwt_expiration'].present?
+      jwt_args[:expiration] = registration_params['jwt_expiration'].to_i.hours.to_i if registration_params['jwt_expiration'] != 'unlimited'
```

Comment:
Do we want to allow unlimited JWTs? From a security perspective that's IMHO a really bad idea. AFAIK there's no way to revoke a JWT other than rotating the signing key which is a very painful operation. If it somehow leaks (copy-paste in some public channel), it becomes a major hassle.

Reply:
Yes we want to, JWT generated for registration are scoped to `RegistrationController` only, so the usage of the token is not unlimited.

Reply:
Please address my comment on how it would be revoked if needed.

Reply:
The only way right now is to do it via rails console, as suggested in [BZ](https://bugzilla.redhat.com/show_bug.cgi?id=1919979#c1):
```
foreman-rake console
User.find_by_login('admin').jwt_secret.destroy
```
That way you get invalidated all the tokens generated by the user.
I created [redmine](https://projects.theforeman.org/issues/32278) task for UI implementation, plus update of docs would be also needed (I'll do it when we have new version of Foreman released.)

Reply:
Thank you, that does address my concerns. I somehow missed that the JWT was always tied to a user and assumed we used JWT in a mode where it's signed with a key. This is likely because that's how they're used in the token-based Puppet certificate workflow.